File: JavaScript Front End/src/Messages/CommandResponseMessages/GraphBuildingFailureResponseMsg.js
const InfoSecCommandResponseMsg = require('./InfoSecCommandResponseMsg.js').InfoSecCommandResponseMsg;

class GraphBuildingFailureResponseMsg extends InfoSecCommandResponseMsg
{
    constructor(jsonMessage)
    {
        super(jsonMessage);
        this.exception = this.jsonMessage["exception"];
    }
}

module.exports = {GraphBuildingFailureResponseMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandResponseMessages/GraphBuildingFailureResponseMsg.java
package InfoSecCooker.Communication.Messages.CommandResponseMessages;

import InfoSecCooker.GraphLogic.GraphBuilding.Exceptions.GraphBuildingException;

public class GraphBuildingFailureResponseMsg extends GraphBuildingCommandResponseMsg
{
    GraphBuildingException exception;

    public GraphBuildingFailureResponseMsg(GraphBuildingException exception, String additionalInfo)
    {
        super(false, additionalInfo);
        this.exception = exception;
    }

    public GraphBuildingFailureResponseMsg(GraphBuildingException exception)
    {
        super(false, "");
        this.exception = exception;
    }
}
